content_scripts/mode_find.js

```diff
@@ -401,5 +402,59 @@ const selectFoundInputElement = function () {
   }
 };
 
+// Retrieve the starting indices of all matches of the queried pattern within the given text.
+const getRegexMatchIndices = (text, regex) => {
+  const indices = [];
+  let match;
+
+  while ((match = regex.exec(text)) !== null) {
+    if (!match[0]) {
+      break;
+    }
+    indices.push(match.index);
+  }
+
+  return indices;
+};
+
+// Highlights text starting from the given startIndex with the specified length.
+const highlight = (textNode, startIndex, length) => {
+  if (startIndex === -1) {
+    return false;
+  }
+  const selection = window.getSelection();
+  const range = document.createRange();
+  range.setStart(textNode, startIndex);
+  range.setEnd(textNode, startIndex + length);
+  selection.removeAllRanges();
+  selection.addRange(range);
+
+  // Ensure the highlighted element is visible within the viewport.
+  const rect = textNode.parentElement.getBoundingClientRect();
+  if (rect.top < 0 || rect.bottom > window.innerHeight) {
+    textNode.parentElement.scrollIntoView({ block: "center" });
+  }
+
+  return true;
+};
+
+const getAllTextNodes = () => {
+  function getAllTextNodes(node, textNodes) {
```

Comment:
This doesn't need to take `textNodes` as a parameter; it can just use the textNodes declared in the surrounding scope. For clarity, move the textNodes declaration on the line above this function declaration.